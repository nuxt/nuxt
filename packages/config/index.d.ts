// Dummy type definition file to provide backward compatibility
// around Nuxt modules extending `@nuxt/config` for Nuxt < 2.9
declare module '@nuxt/config'
