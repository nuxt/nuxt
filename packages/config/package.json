{
  "name": "@nuxt/config",
  "version": "2.15.6",
  "repository": "nuxt/nuxt.js",
  "license": "MIT",
  "main": "dist/config.js",
  "typings": "index.d.ts",
  "files": [
    "dist",
    "index.d.ts"
  ],
  "dependencies": {
    "@nuxt/utils": "2.15.6",
    "consola": "^2.15.3",
    "defu": "^5.0.0",
    "destr": "^1.1.0",
    "dotenv": "^10.0.0",
    "lodash": "^4.17.21",
    "rc9": "^1.2.0",
    "std-env": "^2.3.0",
    "ufo": "^0.7.7"
  },
  "publishConfig": {
    "access": "public"
  }
}
