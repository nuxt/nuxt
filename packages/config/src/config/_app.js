export default () => ({
  vue: {
    config: {
      silent: undefined, // = !dev
      performance: undefined // = dev
    }
  },

  vueMeta: null,

  head: {
    meta: [],
    link: [],
    style: [],
    script: []
  },

  fetch: {
    server: true,
    client: true
  },

  plugins: [],

  extendPlugins: null,

  css: [],

  layouts: {},

  ErrorPage: null,

  loading: {
    color: 'black',
    failedColor: 'red',
    height: '2px',
    throttle: 200,
    duration: 5000,
    continuous: false,
    rtl: false,
    css: true
  },

  loadingIndicator: 'default',

  pageTransition: {
    name: 'page',
    mode: 'out-in',
    appear: false,
    appearClass: 'appear',
    appearActiveClass: 'appear-active',
    appearToClass: 'appear-to'
  },

  layoutTransition: {
    name: 'layout',
    mode: 'out-in'
  },

  features: {
    store: true,
    layouts: true,
    meta: true,
    middleware: true,
    transitions: true,
    deprecations: true,
    validate: true,
    asyncData: true,
    fetch: true,
    clientOnline: true,
    clientPrefetch: true,
    componentAliases: true,
    componentClientOnly: true
  }
})
