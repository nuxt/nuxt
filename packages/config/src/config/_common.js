import { capitalize } from 'lodash'
import env from 'std-env'
import { TARGETS } from '@nuxt/utils'

export default () => ({
  // Env
  dev: Boolean(env.dev),
  test: Boolean(env.test),
  debug: undefined, // = dev
  env: {},

  createRequire: undefined,

  // Target
  target: TARGETS.server,

  // Rendering
  ssr: true,

  // Mode (deprecated)
  mode: undefined,

  // Modern
  modern: undefined,

  // Modules
  modules: [],
  buildModules: [],
  _modules: [],

  globalName: undefined,
  globals: {
    id: globalName => `__${globalName}`,
    nuxt: globalName => `$${globalName}`,
    context: globalName => `__${globalName.toUpperCase()}__`,
    pluginPrefix: globalName => globalName,
    readyCallback: globalName => `on${capitalize(globalName)}Ready`,
    loadedCallback: globalName => `_on${capitalize(globalName)}Loaded`
  },

  // Server
  serverMiddleware: [],

  // Dirs and extensions
  _nuxtConfigFile: undefined,
  srcDir: undefined,
  buildDir: '.nuxt',
  modulesDir: [
    'node_modules'
  ],
  dir: {
    assets: 'assets',
    app: 'app',
    layouts: 'layouts',
    middleware: 'middleware',
    pages: 'pages',
    static: 'static',
    store: 'store'
  },
  extensions: [],
  styleExtensions: ['css', 'pcss', 'postcss', 'styl', 'stylus', 'scss', 'sass', 'less'],
  alias: {},

  // Ignores
  ignoreOptions: undefined,
  ignorePrefix: '-',
  ignore: [
    '**/*.test.*',
    '**/*.spec.*'
  ],

  // Watch
  watch: [],
  watchers: {
    rewatchOnRawEvents: undefined,
    webpack: {
      aggregateTimeout: 1000
    },
    chokidar: {
      ignoreInitial: true
    }
  },

  // Editor
  editor: undefined,

  // Hooks
  hooks: null,

  // runtimeConfig
  privateRuntimeConfig: {},
  publicRuntimeConfig: {}
})
