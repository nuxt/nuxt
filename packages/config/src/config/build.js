import env from 'std-env'

export default () => ({
  quiet: Boolean(env.ci || env.test),
  analyze: false,
  profile: process.argv.includes('--profile'),
  extractCSS: false,
  cssSourceMap: undefined,
  ssr: undefined,
  parallel: false,
  cache: false,
  standalone: false,
  publicPath: '/_nuxt/',
  serverURLPolyfill: 'url',
  filenames: {
    // { isDev, isClient, isServer }
    app: ({ isDev, isModern }) => isDev ? `[name]${isModern ? '.modern' : ''}.js` : `[contenthash:7]${isModern ? '.modern' : ''}.js`,
    chunk: ({ isDev, isModern }) => isDev ? `[name]${isModern ? '.modern' : ''}.js` : `[contenthash:7]${isModern ? '.modern' : ''}.js`,
    css: ({ isDev }) => isDev ? '[name].css' : 'css/[contenthash:7].css',
    img: ({ isDev }) => isDev ? '[path][name].[ext]' : 'img/[name].[contenthash:7].[ext]',
    font: ({ isDev }) => isDev ? '[path][name].[ext]' : 'fonts/[name].[contenthash:7].[ext]',
    video: ({ isDev }) => isDev ? '[path][name].[ext]' : 'videos/[name].[contenthash:7].[ext]'
  },
  loaders: {
    file: { esModule: false },
    fontUrl: { esModule: false, limit: 1000 },
    imgUrl: { esModule: false, limit: 1000 },
    pugPlain: {},
    vue: {
      transformAssetUrls: {
        video: 'src',
        source: 'src',
        object: 'src',
        embed: 'src'
      }
    },
    css: {
      esModule: false,
      modules: {
        compileType: 'icss'
      }
    },
    cssModules: {
      esModule: false,
      modules: {
        localIdentName: '[local]_[hash:base64:5]'
      }
    },
    less: {},
    sass: {
      sassOptions: {
        indentedSyntax: true
      }
    },
    scss: {},
    stylus: {},
    vueStyle: {}
  },
  styleResources: {},
  plugins: [],
  terser: {},
  hardSource: false,
  aggressiveCodeRemoval: false,
  optimizeCSS: undefined,
  optimization: {
    runtimeChunk: 'single',
    minimize: undefined,
    minimizer: undefined,
    splitChunks: {
      chunks: 'all',
      automaticNameDelimiter: '/',
      cacheGroups: {}
    }
  },
  splitChunks: {
    layouts: false,
    pages: true,
    commons: true
  },
  corejs: 'auto',
  babel: {
    configFile: false,
    babelrc: false,
    cacheDirectory: undefined
  },
  transpile: [], // Name of NPM packages to be transpiled
  postcss: {
    preset: {
      // https://cssdb.org/#staging-process
      stage: 2
    }
  },
  html: {
    minify: {
      collapseBooleanAttributes: true,
      decodeEntities: true,
      minifyCSS: true,
      minifyJS: true,
      processConditionalComments: true,
      removeEmptyAttributes: true,
      removeRedundantAttributes: true,
      trimCustomFragments: true,
      useShortDoctype: true
    }
  },

  template: undefined,
  templates: [],

  watch: [],
  devMiddleware: {
    // stats will be printed by webapckbar StateReporter
    stats: 'none'
  },
  hotMiddleware: {},

  stats: {
    excludeAssets: [
      /.map$/,
      /index\..+\.html$/,
      /vue-ssr-(client|modern)-manifest.json/
    ]
  },
  friendlyErrors: true,
  additionalExtensions: [],
  warningIgnoreFilters: [],

  followSymlinks: false,

  loadingScreen: {},
  indicator: {
    position: 'bottom-right',
    backgroundColor: '#2E495E',
    color: '#00C48D'
  }
})
