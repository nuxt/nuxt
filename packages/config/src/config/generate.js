export default () => ({
  dir: 'dist',
  routes: [],
  exclude: [],
  concurrency: 500,
  ignoreEnv: false,
  interval: 0,
  subFolders: true,
  fallback: '200.html',
  crawler: true,
  manifest: true,
  nojekyll: true,
  cache: {
    ignore: [],
    globbyOptions: {
      gitignore: true
    }
  },
  staticAssets: {
    base: undefined, // Default: "/_nuxt/static:
    versionBase: undefined, // Default: "_nuxt/static/{version}""
    dir: 'static',
    version: undefined // Default: "{timeStampSec}"
  }
})
