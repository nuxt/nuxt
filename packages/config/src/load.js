import path from 'path'
import fs from 'fs'
import defu from 'defu'
import consola from 'consola'
import dotenv from 'dotenv'
import { clearRequireCache, createRequire, scanRequireTree } from '@nuxt/utils'
import destr from 'destr'
import * as rc from 'rc9'
import { defaultNuxtConfigFile } from './config'

export async function loadNuxtConfig ({
  rootDir = '.',
  envConfig = {},
  configFile = defaultNuxtConfigFile,
  configContext = {},
  configOverrides = {}
} = {}) {
  rootDir = path.resolve(rootDir)

  const _require = createRequire(rootDir, true)

  let options = {}

  try {
    configFile = _require.resolve(path.resolve(rootDir, configFile))
  } catch (e) {
    if (e.code !== 'MODULE_NOT_FOUND') {
      throw (e)
    } else if (configFile !== defaultNuxtConfigFile) {
      consola.fatal('Config file not found: ' + configFile)
    }
    // Skip configFile if cannot resolve
    configFile = undefined
  }

  // Load env
  envConfig = {
    dotenv: '.env',
    env: process.env,
    expand: true,
    ...envConfig
  }

  const env = loadEnv(envConfig, rootDir)

  // Fill process.env so it is accessible in nuxt.config
  for (const key in env) {
    if (!key.startsWith('_') && envConfig.env[key] === undefined) {
      envConfig.env[key] = env[key]
    }
  }

  if (configFile) {
    // Clear cache
    clearRequireCache(configFile)
    options = _require(configFile) || {}
    if (options.default) {
      options = options.default
    }

    if (typeof options === 'function') {
      try {
        options = await options(configContext)
        if (options.default) {
          options = options.default
        }
      } catch (error) {
        consola.error(error)
        consola.fatal('Error while fetching async configuration')
      }
    }

    // Don't mutate options export
    options = { ...options }

    // Keep _nuxtConfigFile for watching
    options._nuxtConfigFile = configFile

    // Keep all related files for watching
    options._nuxtConfigFiles = Array.from(scanRequireTree(configFile))
    if (!options._nuxtConfigFiles.includes(configFile)) {
      options._nuxtConfigFiles.unshift(configFile)
    }
  }

  if (typeof options.rootDir !== 'string') {
    options.rootDir = rootDir
  }

  // Load Combine configs
  // Priority: configOverrides > nuxtConfig > .nuxt/dist/nuxtrc > .nuxtrc > .nuxtrc (global)
  const dev = configOverrides.dev ?? options.dev ?? configContext.dev
  const buildDir = configOverrides.buildDir ?? options.buildDir ?? configContext.buildDir ?? '.nuxt'
  options = defu(
    configOverrides,
    options,
    !dev ? rc.read({ name: 'nuxtrc', dir: path.resolve(buildDir, 'dist') }) : {},
    rc.read({ name: '.nuxtrc', dir: options.rootDir }),
    rc.readUser('.nuxtrc')
  )

  // Load env to options._env
  options._env = env
  options._envConfig = envConfig
  if (configContext) { configContext.env = env }

  // Expand and interpolate runtimeConfig from _env
  if (envConfig.expand) {
    for (const c of ['publicRuntimeConfig', 'privateRuntimeConfig']) {
      if (options[c]) {
        if (typeof options[c] === 'function') {
          options[c] = options[c](env)
        }
        expand(options[c], env, destr)
      }
    }
  }

  return options
}

function loadEnv (envConfig, rootDir = process.cwd()) {
  const env = Object.create(null)

  // Read dotenv
  if (envConfig.dotenv) {
    envConfig.dotenv = path.resolve(rootDir, envConfig.dotenv)
    if (fs.existsSync(envConfig.dotenv)) {
      const parsed = dotenv.parse(fs.readFileSync(envConfig.dotenv, 'utf-8'))
      Object.assign(env, parsed)
    }
  }

  // Apply process.env
  if (!envConfig.env._applied) {
    Object.assign(env, envConfig.env)
    envConfig.env._applied = true
  }

  // Interpolate env
  if (envConfig.expand) {
    expand(env)
  }

  return env
}

// Based on https://github.com/motdotla/dotenv-expand
function expand (target, source = {}, parse = v => v) {
  function getValue (key) {
    // Source value 'wins' over target value
    return source[key] !== undefined ? source[key] : target[key]
  }

  function interpolate (value, parents = []) {
    if (typeof value !== 'string') {
      return value
    }
    const matches = value.match(/(.?\${?(?:[a-zA-Z0-9_:]+)?}?)/g) || []
    return parse(matches.reduce((newValue, match) => {
      const parts = /(.?)\${?([a-zA-Z0-9_:]+)?}?/g.exec(match)
      const prefix = parts[1]

      let value, replacePart

      if (prefix === '\\') {
        replacePart = parts[0]
        value = replacePart.replace('\\$', '$')
      } else {
        const key = parts[2]
        replacePart = parts[0].substring(prefix.length)

        // Avoid recursion
        if (parents.includes(key)) {
          consola.warn(`Please avoid recursive environment variables ( loop: ${parents.join(' > ')} > ${key} )`)
          return ''
        }

        value = getValue(key)

        // Resolve recursive interpolations
        value = interpolate(value, [...parents, key])
      }

      return value !== undefined ? newValue.replace(replacePart, value) : newValue
    }, value))
  }

  for (const key in target) {
    target[key] = interpolate(getValue(key))
  }
}
