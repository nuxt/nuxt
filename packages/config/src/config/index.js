import _app from './_app'
import _common from './_common'

import build from './build'
import messages from './messages'
import modes from './modes'
import render from './render'
import router from './router'
import server from './server'
import cli from './cli'
import generate from './generate'

export const defaultNuxtConfigFile = 'nuxt.config'

export function getDefaultNuxtConfig (options = {}) {
  if (!options.env) {
    options.env = process.env
  }

  return {
    ..._app(),
    ..._common(),
    build: build(),
    messages: messages(),
    modes: modes(),
    render: render(),
    router: router(),
    server: server(options),
    cli: cli(),
    generate: generate()
  }
}
