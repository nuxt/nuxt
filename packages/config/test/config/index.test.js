import { getDefaultNuxtConfig } from '../../src/config'

jest.mock('std-env', () => ({
  isBrowser: false,
  isTest: true,
  isDevelopment: false,
  isProduction: true,
  isDebug: false,
  isCI: true,
  isWindows: false,
  isMacOS: false,
  isLinux: true
}))

describe('config', () => {
  test('should return default nuxt configurations', () => {
    expect(getDefaultNuxtConfig()).toMatchSnapshot()
  })

  test('should return nuxt configurations with custom env', () => {
    const env = {
      NUXT_PORT: '3001',
      NUXT_HOST: 'localhost',
      UNIX_SOCKET: '/var/run/nuxt.sock'
    }
    const config = getDefaultNuxtConfig({ env })
    expect(config).toMatchSnapshot()
  })
})
