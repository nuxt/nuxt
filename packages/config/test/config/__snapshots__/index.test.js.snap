// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should return default nuxt configurations 1`] = `
Object {
  "ErrorPage": null,
  "_modules": Array [],
  "_nuxtConfigFile": undefined,
  "alias": Object {},
  "build": Object {
    "additionalExtensions": Array [],
    "aggressiveCodeRemoval": false,
    "analyze": false,
    "babel": Object {
      "babelrc": false,
      "cacheDirectory": undefined,
      "configFile": false,
    },
    "cache": false,
    "corejs": "auto",
    "cssSourceMap": undefined,
    "devMiddleware": Object {
      "stats": "none",
    },
    "extractCSS": false,
    "filenames": Object {
      "app": [Function],
      "chunk": [Function],
      "css": [Function],
      "font": [Function],
      "img": [Function],
      "video": [Function],
    },
    "followSymlinks": false,
    "friendlyErrors": true,
    "hardSource": false,
    "hotMiddleware": Object {},
    "html": Object {
      "minify": Object {
        "collapseBooleanAttributes": true,
        "decodeEntities": true,
        "minifyCSS": true,
        "minifyJS": true,
        "processConditionalComments": true,
        "removeEmptyAttributes": true,
        "removeRedundantAttributes": true,
        "trimCustomFragments": true,
        "useShortDoctype": true,
      },
    },
    "indicator": Object {
      "backgroundColor": "#2E495E",
      "color": "#00C48D",
      "position": "bottom-right",
    },
    "loaders": Object {
      "css": Object {
        "esModule": false,
        "modules": Object {
          "compileType": "icss",
        },
      },
      "cssModules": Object {
        "esModule": false,
        "modules": Object {
          "localIdentName": "[local]_[hash:base64:5]",
        },
      },
      "file": Object {
        "esModule": false,
      },
      "fontUrl": Object {
        "esModule": false,
        "limit": 1000,
      },
      "imgUrl": Object {
        "esModule": false,
        "limit": 1000,
      },
      "less": Object {},
      "pugPlain": Object {},
      "sass": Object {
        "sassOptions": Object {
          "indentedSyntax": true,
        },
      },
      "scss": Object {},
      "stylus": Object {},
      "vue": Object {
        "transformAssetUrls": Object {
          "embed": "src",
          "object": "src",
          "source": "src",
          "video": "src",
        },
      },
      "vueStyle": Object {},
    },
    "loadingScreen": Object {},
    "optimization": Object {
      "minimize": undefined,
      "minimizer": undefined,
      "runtimeChunk": "single",
      "splitChunks": Object {
        "automaticNameDelimiter": "/",
        "cacheGroups": Object {},
        "chunks": "all",
      },
    },
    "optimizeCSS": undefined,
    "parallel": false,
    "plugins": Array [],
    "postcss": Object {
      "preset": Object {
        "stage": 2,
      },
    },
    "profile": false,
    "publicPath": "/_nuxt/",
    "quiet": true,
    "serverURLPolyfill": "url",
    "splitChunks": Object {
      "commons": true,
      "layouts": false,
      "pages": true,
    },
    "ssr": undefined,
    "standalone": false,
    "stats": Object {
      "excludeAssets": Array [
        /\\.map\\$/,
        /index\\\\\\.\\.\\+\\\\\\.html\\$/,
        /vue-ssr-\\(client\\|modern\\)-manifest\\.json/,
      ],
    },
    "styleResources": Object {},
    "template": undefined,
    "templates": Array [],
    "terser": Object {},
    "transpile": Array [],
    "warningIgnoreFilters": Array [],
    "watch": Array [],
  },
  "buildDir": ".nuxt",
  "buildModules": Array [],
  "cli": Object {
    "badgeMessages": Array [],
    "bannerColor": "green",
  },
  "createRequire": undefined,
  "css": Array [],
  "debug": undefined,
  "dev": false,
  "dir": Object {
    "app": "app",
    "assets": "assets",
    "layouts": "layouts",
    "middleware": "middleware",
    "pages": "pages",
    "static": "static",
    "store": "store",
  },
  "editor": undefined,
  "env": Object {},
  "extendPlugins": null,
  "extensions": Array [],
  "features": Object {
    "asyncData": true,
    "clientOnline": true,
    "clientPrefetch": true,
    "componentAliases": true,
    "componentClientOnly": true,
    "deprecations": true,
    "fetch": true,
    "layouts": true,
    "meta": true,
    "middleware": true,
    "store": true,
    "transitions": true,
    "validate": true,
  },
  "fetch": Object {
    "client": true,
    "server": true,
  },
  "generate": Object {
    "cache": Object {
      "globbyOptions": Object {
        "gitignore": true,
      },
      "ignore": Array [],
    },
    "concurrency": 500,
    "crawler": true,
    "dir": "dist",
    "exclude": Array [],
    "fallback": "200.html",
    "ignoreEnv": false,
    "interval": 0,
    "manifest": true,
    "nojekyll": true,
    "routes": Array [],
    "staticAssets": Object {
      "base": undefined,
      "dir": "static",
      "version": undefined,
      "versionBase": undefined,
    },
    "subFolders": true,
  },
  "globalName": undefined,
  "globals": Object {
    "context": [Function],
    "id": [Function],
    "loadedCallback": [Function],
    "nuxt": [Function],
    "pluginPrefix": [Function],
    "readyCallback": [Function],
  },
  "head": Object {
    "link": Array [],
    "meta": Array [],
    "script": Array [],
    "style": Array [],
  },
  "hooks": null,
  "ignore": Array [
    "**/*.test.*",
    "**/*.spec.*",
  ],
  "ignoreOptions": undefined,
  "ignorePrefix": "-",
  "layoutTransition": Object {
    "mode": "out-in",
    "name": "layout",
  },
  "layouts": Object {},
  "loading": Object {
    "color": "black",
    "continuous": false,
    "css": true,
    "duration": 5000,
    "failedColor": "red",
    "height": "2px",
    "rtl": false,
    "throttle": 200,
  },
  "loadingIndicator": "default",
  "messages": Object {
    "back_to_home": "Back to the home page",
    "client_error": "Error",
    "client_error_details": "An error occurred while rendering the page. Check developer tools console for details.",
    "error_404": "This page could not be found",
    "loading": "Loading...",
    "nuxtjs": "Nuxt",
    "server_error": "Server error",
    "server_error_details": "An error occurred in the application and your page could not be served. If you are the application owner, check your logs for details.",
  },
  "mode": undefined,
  "modern": undefined,
  "modes": Object {
    "spa": Object {
      "build": Object {
        "ssr": false,
      },
      "render": Object {
        "ssr": false,
      },
    },
    "universal": Object {
      "build": Object {
        "ssr": true,
      },
      "render": Object {
        "ssr": true,
      },
    },
  },
  "modules": Array [],
  "modulesDir": Array [
    "node_modules",
  ],
  "pageTransition": Object {
    "appear": false,
    "appearActiveClass": "appear-active",
    "appearClass": "appear",
    "appearToClass": "appear-to",
    "mode": "out-in",
    "name": "page",
  },
  "plugins": Array [],
  "privateRuntimeConfig": Object {},
  "publicRuntimeConfig": Object {},
  "render": Object {
    "bundleRenderer": Object {
      "runInNewContext": undefined,
      "shouldPrefetch": [Function],
      "shouldPreload": [Function],
    },
    "compressor": Object {
      "threshold": 0,
    },
    "crossorigin": undefined,
    "csp": false,
    "dist": Object {
      "index": false,
      "maxAge": "1y",
    },
    "etag": Object {
      "weak": false,
    },
    "fallback": Object {
      "dist": Object {},
      "static": Object {
        "handlers": Object {
          ".htm": false,
          ".html": false,
        },
        "skipUnknown": true,
      },
    },
    "http2": Object {
      "push": false,
      "pushAssets": null,
      "shouldPush": null,
    },
    "resourceHints": true,
    "ssr": undefined,
    "ssrLog": undefined,
    "static": Object {
      "prefix": true,
    },
  },
  "router": Object {
    "base": "/",
    "extendRoutes": null,
    "fallback": false,
    "linkActiveClass": "nuxt-link-active",
    "linkExactActiveClass": "nuxt-link-exact-active",
    "linkPrefetchedClass": false,
    "middleware": Array [],
    "mode": "history",
    "parseQuery": false,
    "prefetchLinks": true,
    "prefetchPayloads": true,
    "routeNameSplitter": "-",
    "routes": Array [],
    "scrollBehavior": null,
    "stringifyQuery": false,
    "trailingSlash": undefined,
  },
  "server": Object {
    "host": "localhost",
    "https": false,
    "port": 3000,
    "socket": undefined,
    "timing": false,
  },
  "serverMiddleware": Array [],
  "srcDir": undefined,
  "ssr": true,
  "styleExtensions": Array [
    "css",
    "pcss",
    "postcss",
    "styl",
    "stylus",
    "scss",
    "sass",
    "less",
  ],
  "target": "server",
  "test": true,
  "vue": Object {
    "config": Object {
      "performance": undefined,
      "silent": undefined,
    },
  },
  "vueMeta": null,
  "watch": Array [],
  "watchers": Object {
    "chokidar": Object {
      "ignoreInitial": true,
    },
    "rewatchOnRawEvents": undefined,
    "webpack": Object {
      "aggregateTimeout": 1000,
    },
  },
}
`;

exports[`config should return nuxt configurations with custom env 1`] = `
Object {
  "ErrorPage": null,
  "_modules": Array [],
  "_nuxtConfigFile": undefined,
  "alias": Object {},
  "build": Object {
    "additionalExtensions": Array [],
    "aggressiveCodeRemoval": false,
    "analyze": false,
    "babel": Object {
      "babelrc": false,
      "cacheDirectory": undefined,
      "configFile": false,
    },
    "cache": false,
    "corejs": "auto",
    "cssSourceMap": undefined,
    "devMiddleware": Object {
      "stats": "none",
    },
    "extractCSS": false,
    "filenames": Object {
      "app": [Function],
      "chunk": [Function],
      "css": [Function],
      "font": [Function],
      "img": [Function],
      "video": [Function],
    },
    "followSymlinks": false,
    "friendlyErrors": true,
    "hardSource": false,
    "hotMiddleware": Object {},
    "html": Object {
      "minify": Object {
        "collapseBooleanAttributes": true,
        "decodeEntities": true,
        "minifyCSS": true,
        "minifyJS": true,
        "processConditionalComments": true,
        "removeEmptyAttributes": true,
        "removeRedundantAttributes": true,
        "trimCustomFragments": true,
        "useShortDoctype": true,
      },
    },
    "indicator": Object {
      "backgroundColor": "#2E495E",
      "color": "#00C48D",
      "position": "bottom-right",
    },
    "loaders": Object {
      "css": Object {
        "esModule": false,
        "modules": Object {
          "compileType": "icss",
        },
      },
      "cssModules": Object {
        "esModule": false,
        "modules": Object {
          "localIdentName": "[local]_[hash:base64:5]",
        },
      },
      "file": Object {
        "esModule": false,
      },
      "fontUrl": Object {
        "esModule": false,
        "limit": 1000,
      },
      "imgUrl": Object {
        "esModule": false,
        "limit": 1000,
      },
      "less": Object {},
      "pugPlain": Object {},
      "sass": Object {
        "sassOptions": Object {
          "indentedSyntax": true,
        },
      },
      "scss": Object {},
      "stylus": Object {},
      "vue": Object {
        "transformAssetUrls": Object {
          "embed": "src",
          "object": "src",
          "source": "src",
          "video": "src",
        },
      },
      "vueStyle": Object {},
    },
    "loadingScreen": Object {},
    "optimization": Object {
      "minimize": undefined,
      "minimizer": undefined,
      "runtimeChunk": "single",
      "splitChunks": Object {
        "automaticNameDelimiter": "/",
        "cacheGroups": Object {},
        "chunks": "all",
      },
    },
    "optimizeCSS": undefined,
    "parallel": false,
    "plugins": Array [],
    "postcss": Object {
      "preset": Object {
        "stage": 2,
      },
    },
    "profile": false,
    "publicPath": "/_nuxt/",
    "quiet": true,
    "serverURLPolyfill": "url",
    "splitChunks": Object {
      "commons": true,
      "layouts": false,
      "pages": true,
    },
    "ssr": undefined,
    "standalone": false,
    "stats": Object {
      "excludeAssets": Array [
        /\\.map\\$/,
        /index\\\\\\.\\.\\+\\\\\\.html\\$/,
        /vue-ssr-\\(client\\|modern\\)-manifest\\.json/,
      ],
    },
    "styleResources": Object {},
    "template": undefined,
    "templates": Array [],
    "terser": Object {},
    "transpile": Array [],
    "warningIgnoreFilters": Array [],
    "watch": Array [],
  },
  "buildDir": ".nuxt",
  "buildModules": Array [],
  "cli": Object {
    "badgeMessages": Array [],
    "bannerColor": "green",
  },
  "createRequire": undefined,
  "css": Array [],
  "debug": undefined,
  "dev": false,
  "dir": Object {
    "app": "app",
    "assets": "assets",
    "layouts": "layouts",
    "middleware": "middleware",
    "pages": "pages",
    "static": "static",
    "store": "store",
  },
  "editor": undefined,
  "env": Object {},
  "extendPlugins": null,
  "extensions": Array [],
  "features": Object {
    "asyncData": true,
    "clientOnline": true,
    "clientPrefetch": true,
    "componentAliases": true,
    "componentClientOnly": true,
    "deprecations": true,
    "fetch": true,
    "layouts": true,
    "meta": true,
    "middleware": true,
    "store": true,
    "transitions": true,
    "validate": true,
  },
  "fetch": Object {
    "client": true,
    "server": true,
  },
  "generate": Object {
    "cache": Object {
      "globbyOptions": Object {
        "gitignore": true,
      },
      "ignore": Array [],
    },
    "concurrency": 500,
    "crawler": true,
    "dir": "dist",
    "exclude": Array [],
    "fallback": "200.html",
    "ignoreEnv": false,
    "interval": 0,
    "manifest": true,
    "nojekyll": true,
    "routes": Array [],
    "staticAssets": Object {
      "base": undefined,
      "dir": "static",
      "version": undefined,
      "versionBase": undefined,
    },
    "subFolders": true,
  },
  "globalName": undefined,
  "globals": Object {
    "context": [Function],
    "id": [Function],
    "loadedCallback": [Function],
    "nuxt": [Function],
    "pluginPrefix": [Function],
    "readyCallback": [Function],
  },
  "head": Object {
    "link": Array [],
    "meta": Array [],
    "script": Array [],
    "style": Array [],
  },
  "hooks": null,
  "ignore": Array [
    "**/*.test.*",
    "**/*.spec.*",
  ],
  "ignoreOptions": undefined,
  "ignorePrefix": "-",
  "layoutTransition": Object {
    "mode": "out-in",
    "name": "layout",
  },
  "layouts": Object {},
  "loading": Object {
    "color": "black",
    "continuous": false,
    "css": true,
    "duration": 5000,
    "failedColor": "red",
    "height": "2px",
    "rtl": false,
    "throttle": 200,
  },
  "loadingIndicator": "default",
  "messages": Object {
    "back_to_home": "Back to the home page",
    "client_error": "Error",
    "client_error_details": "An error occurred while rendering the page. Check developer tools console for details.",
    "error_404": "This page could not be found",
    "loading": "Loading...",
    "nuxtjs": "Nuxt",
    "server_error": "Server error",
    "server_error_details": "An error occurred in the application and your page could not be served. If you are the application owner, check your logs for details.",
  },
  "mode": undefined,
  "modern": undefined,
  "modes": Object {
    "spa": Object {
      "build": Object {
        "ssr": false,
      },
      "render": Object {
        "ssr": false,
      },
    },
    "universal": Object {
      "build": Object {
        "ssr": true,
      },
      "render": Object {
        "ssr": true,
      },
    },
  },
  "modules": Array [],
  "modulesDir": Array [
    "node_modules",
  ],
  "pageTransition": Object {
    "appear": false,
    "appearActiveClass": "appear-active",
    "appearClass": "appear",
    "appearToClass": "appear-to",
    "mode": "out-in",
    "name": "page",
  },
  "plugins": Array [],
  "privateRuntimeConfig": Object {},
  "publicRuntimeConfig": Object {},
  "render": Object {
    "bundleRenderer": Object {
      "runInNewContext": undefined,
      "shouldPrefetch": [Function],
      "shouldPreload": [Function],
    },
    "compressor": Object {
      "threshold": 0,
    },
    "crossorigin": undefined,
    "csp": false,
    "dist": Object {
      "index": false,
      "maxAge": "1y",
    },
    "etag": Object {
      "weak": false,
    },
    "fallback": Object {
      "dist": Object {},
      "static": Object {
        "handlers": Object {
          ".htm": false,
          ".html": false,
        },
        "skipUnknown": true,
      },
    },
    "http2": Object {
      "push": false,
      "pushAssets": null,
      "shouldPush": null,
    },
    "resourceHints": true,
    "ssr": undefined,
    "ssrLog": undefined,
    "static": Object {
      "prefix": true,
    },
  },
  "router": Object {
    "base": "/",
    "extendRoutes": null,
    "fallback": false,
    "linkActiveClass": "nuxt-link-active",
    "linkExactActiveClass": "nuxt-link-exact-active",
    "linkPrefetchedClass": false,
    "middleware": Array [],
    "mode": "history",
    "parseQuery": false,
    "prefetchLinks": true,
    "prefetchPayloads": true,
    "routeNameSplitter": "-",
    "routes": Array [],
    "scrollBehavior": null,
    "stringifyQuery": false,
    "trailingSlash": undefined,
  },
  "server": Object {
    "host": "localhost",
    "https": false,
    "port": "3001",
    "socket": "/var/run/nuxt.sock",
    "timing": false,
  },
  "serverMiddleware": Array [],
  "srcDir": undefined,
  "ssr": true,
  "styleExtensions": Array [
    "css",
    "pcss",
    "postcss",
    "styl",
    "stylus",
    "scss",
    "sass",
    "less",
  ],
  "target": "server",
  "test": true,
  "vue": Object {
    "config": Object {
      "performance": undefined,
      "silent": undefined,
    },
  },
  "vueMeta": null,
  "watch": Array [],
  "watchers": Object {
    "chokidar": Object {
      "ignoreInitial": true,
    },
    "rewatchOnRawEvents": undefined,
    "webpack": Object {
      "aggregateTimeout": 1000,
    },
  },
}
`;
